### ISSUE TITLE
Permission error when switching from fabric8-maven-plugin to openshift-maven-plugin

### ISSUE THREAD

**Issue by ochstobi** [eclipse/jkube#581]
### Info

* Eclipse JKube version : openshift-maven-plugin version: 1.1.0
* Maven version (`mvn -v`) : 3.6.1
* Kubernetes / Red Hat OpenShift setup and version : Kubernetes v1.18.3+ac53d20 / Openshift 4

If I try to apply resources with the openshift-maven-plugin I get this error:

```
[INFO] --- openshift-maven-plugin:1.1.0:apply (docker-local-resource-apply) @ helloworld.openshift ---
[INFO] oc: Using OpenShift at https://myserver/ in namespace mynamespace with manifest C:\..\target\classes\META-INF\jkube\openshift.yml
[INFO] oc: OpenShift platform detected
[INFO] oc: Applying Project mynamespace from openshift.yml
[ERROR] oc: Failed to create Project from openshift.yml. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://myserver/apis/project.openshift.io/v1/projects. Message: Forbidden! User MYUSER/myserver doesn't have permission. projects.project.openshift.io is forbidden: User "MYUSER" cannot create resource "projects" in API group "project.openshift.io" at the cluster scope.: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://myserver/apis/project.openshift.io/v1/projects. Message: Forbidden! User MYUSER/myserver doesn't have permission. projects.project.openshift.io is forbidden: User "MYUSER" cannot create resource "projects" in API group "project.openshift.io" at the cluster scope.
```
If I apply the same resources with fabric8-maven-plugin:4.4.1 it is working. So I don't think this is a permission issue. How can I debug this further to get rid of this?

I found out that in the target folder the generated openshift.yml now has the following entry which is causing the error:
```
- apiVersion: project.openshift.io/v1
  kind: Project
  metadata:
    labels:
      app: helloworld.openshift
      provider: jkube
      version: 12.1.1
      group: helloworld
    name: namespace
  status:
    phase: Active
```
But I haven't figured out where this is from.

Maven log with -X:
```
[ERROR] Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.1.0:apply (docker-local-resource-apply) on project helloworld.openshift: Failed to create Project from openshift.yml. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://myserver/apis/project.openshift.io/v1/projects. Message: Forbidden! User myuser/myserver doesn't have permission. projects.project.openshift.io is forbidden: User "myuser" cannot create resource "projects" in API group "project.openshift.io" at the cluster scope. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.1.0:apply (docker-local-resource-apply) on project helloworld.openshift: Failed to create Project from openshift.yml. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://myserver/apis/project.openshift.io/v1/projects. Message: Forbidden! User myuser/myserver doesn't have permission. projects.project.openshift.io is forbidden: User "myuser" cannot create resource "projects" in API group "project.openshift.io" at the cluster scope.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to create Project from openshift.yml. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://myserver/apis/project.openshift.io/v1/projects. Message: Forbidden! User myuser/myserver doesn't have permission. projects.project.openshift.io is forbidden: User "myuser" cannot create resource "projects" in API group "project.openshift.io" at the cluster scope.
    at org.eclipse.jkube.maven.plugin.mojo.build.ApplyMojo.executeInternal (ApplyMojo.java:192)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractJKubeMojo.execute (AbstractJKubeMojo.java:92)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.RuntimeException: Failed to create Project from openshift.yml. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://myserver/apis/project.openshift.io/v1/projects. Message: Forbidden! User myuser/myserver doesn't have permission. projects.project.openshift.io is forbidden: User "myuser" cannot create resource "projects" in API group "project.openshift.io" at the cluster scope.
    at org.eclipse.jkube.kit.config.service.ApplyService.onApplyError (ApplyService.java:1301)
    at org.eclipse.jkube.kit.config.service.ApplyService.applyEntity (ApplyService.java:204)
    at org.eclipse.jkube.kit.config.service.ApplyService.apply (ApplyService.java:136)
    at org.eclipse.jkube.kit.config.service.ApplyService.applyStandardEntities (ApplyService.java:1432)
    at org.eclipse.jkube.kit.config.service.ApplyService.applyEntities (ApplyService.java:1406)
    at org.eclipse.jkube.maven.plugin.mojo.build.ApplyMojo.applyEntities (ApplyMojo.java:198)
    at org.eclipse.jkube.maven.plugin.mojo.build.ApplyMojo.executeInternal (ApplyMojo.java:185)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractJKubeMojo.execute (AbstractJKubeMojo.java:92)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://myserver/apis/project.openshift.io/v1/projects. Message: Forbidden! User myuser/myserver doesn't have permission. projects.project.openshift.io is forbidden: User "myuser" cannot create resource "projects" in API group "project.openshift.io" at the cluster scope.
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure (OperationSupport.java:570)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode (OperationSupport.java:507)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse (OperationSupport.java:474)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse (OperationSupport.java:435)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate (OperationSupport.java:250)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate (BaseOperation.java:871)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create (BaseOperation.java:351)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create (BaseOperation.java:85)
    at io.fabric8.openshift.client.handlers.ProjectHandler.create (ProjectHandler.java:42)
    at io.fabric8.openshift.client.handlers.ProjectHandler.create (ProjectHandler.java:28)
    at io.fabric8.kubernetes.client.utils.CreateOrReplaceHelper.lambda$createOrReplaceItem$0 (CreateOrReplaceHelper.java:74)
    at io.fabric8.kubernetes.client.utils.CreateOrReplaceHelper$$Lambda$175/0000000000000000.apply (Unknown Source)
    at io.fabric8.kubernetes.client.utils.CreateOrReplaceHelper.createOrReplace (CreateOrReplaceHelper.java:53)
    at io.fabric8.kubernetes.client.utils.CreateOrReplaceHelper.createOrReplaceItem (CreateOrReplaceHelper.java:87)
    at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.createOrReplace (NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java:142)
    at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.createOrReplace (NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java:61)
    at org.eclipse.jkube.kit.config.service.ApplyService.applyEntity (ApplyService.java:202)
    at org.eclipse.jkube.kit.config.service.ApplyService.apply (ApplyService.java:136)
    at org.eclipse.jkube.kit.config.service.ApplyService.applyStandardEntities (ApplyService.java:1432)
    at org.eclipse.jkube.kit.config.service.ApplyService.applyEntities (ApplyService.java:1406)
    at org.eclipse.jkube.maven.plugin.mojo.build.ApplyMojo.applyEntities (ApplyMojo.java:198)
    at org.eclipse.jkube.maven.plugin.mojo.build.ApplyMojo.executeInternal (ApplyMojo.java:185)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractJKubeMojo.execute (AbstractJKubeMojo.java:92)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

Best regards.

**Comment by manusa** [comment]
This is caused by https://github.com/eclipse/jkube/issues/570

I guess you configured your namespace using the `jkube.namespace` and a project resource is being created despite you already have that project already in your cluster (you just wanted to configure it), right?

We're working on a fix for this.

**Comment by ochstobi** [comment]
Thats excatly my problem. Haven't seen that issue, I'm sorry for duplicate.

**Comment by manusa** [comment]
No worries.
We'll probably fix this in 1.1 since it might affect most users running JKube in a Pipeline.
